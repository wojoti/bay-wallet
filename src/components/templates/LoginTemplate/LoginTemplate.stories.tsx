import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import LoginTemplate from './LoginTemplate';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Templates/LoginTemplate',
  component: LoginTemplate,
  argTypes: {
    onLoginSubmit: {action: 'onLoginSubmit'},
    onPasswordReset: {action: 'onPasswordReset'},
    onSocialLoginPress: {action: 'onSocialLoginPress'},
  },
} as ComponentMeta<typeof LoginTemplate>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof LoginTemplate> = args => (
  <LoginTemplate {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
