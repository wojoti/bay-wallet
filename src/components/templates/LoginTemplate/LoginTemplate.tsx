import {Container, Img, ImgType} from 'components/atoms';
import {LoginForm} from 'components/organisms';
import {PropsWithChildren} from 'react';
import style from './LoginTemplate.style';

export type LoginTemplateProps = PropsWithChildren<{
  onLoginSubmit: (username: string, password: string) => void;
  onPasswordReset: () => void;
  onSocialLoginPress: (icon: string) => void;
  testId?: string;
}>;

const LoginTemplate = ({
  onLoginSubmit,
  onPasswordReset,
  onSocialLoginPress,
  testId,
}: LoginTemplateProps) => {
  const onSocialPress = () => {
    onSocialLoginPress(
      'beda 2 funkcje od fejsa i googla - poki co nie mam komponentow',
    );
  };
  return (
    <Container testId={testId} customStyle={style.wrapper}>
      <Container flex={1} justify="center" align="center">
        <Container flex={1} justify="flex-end">
          <Img
            src={ImgType.logoSplashWhite}
            height={230}
            alt={''}
            testId={'logintemplate-img'}
          />
        </Container>
        <Container flex={2} customStyle={style.sizer}>
          <LoginForm
            onLoginSubmit={onLoginSubmit}
            onPasswordReset={onPasswordReset}
          />
        </Container>
      </Container>
    </Container>
  );
};

export default LoginTemplate;
