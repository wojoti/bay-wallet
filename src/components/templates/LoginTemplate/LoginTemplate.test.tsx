import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import LoginTemplate, {LoginTemplateProps} from './LoginTemplate';

const mockOnLoginSubmit = jest.fn<
  LoginTemplateProps['onLoginSubmit'],
  [email: string, password: string]
>();
const handleResetPress = jest.fn();
const handleSocialLoginPress = jest.fn<
  LoginTemplateProps['onPasswordReset'],
  [icon: string]
>();
const props: LoginTemplateProps = {
  onLoginSubmit: mockOnLoginSubmit,
  onPasswordReset: handleResetPress,
  onSocialLoginPress: handleSocialLoginPress,
  testId: 'test-logintemplate-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<LoginTemplate {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render logintemplate', () => {
  render(<LoginTemplate {...props} />);
  const logintemplateElement = screen.getByTestId('test-logintemplate-id');
  expect(logintemplateElement).toBeOnTheScreen();
});
