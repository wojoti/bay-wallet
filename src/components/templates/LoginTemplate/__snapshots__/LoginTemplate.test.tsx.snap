// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "backgroundColor": "white",
        "height": "100%",
      },
    ]
  }
  testID="test-logintemplate-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <SvgMock
        height={230}
        testID="logintemplate-img"
        width="100%"
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": 2,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "width": "85%",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": undefined,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontFamily": "Poppins-Bold",
              "fontSize": 24,
              "textAlign": "left",
            }
          }
          testID="loginform-header"
        >
          Login
        </Text>
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": undefined,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": "100%",
              },
              undefined,
            ]
          }
          testID="loginform-inputarea-username"
        >
          <View
            style={
              [
                {
                  "alignItems": "stretch",
                  "flex": undefined,
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "width": "100%",
                },
                {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "flex": 0,
                  "marginHorizontal": 15,
                  "paddingHorizontal": 10,
                  "top": 15,
                  "width": 95,
                  "zIndex": 1,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#363062",
                  "fontFamily": "Poppins-Light",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "textAlign": "left",
                }
              }
              testID="inputarea-label"
            >
              Username
            </Text>
          </View>
          <TextInput
            accessibilityHint="Allow to insert username"
            accessibilityLabel="Login input field"
            autoComplete="username"
            autoCorrect={false}
            onChangeText={[Function]}
            style={
              {
                "borderColor": "#363062",
                "borderRadius": 8,
                "borderWidth": 1,
                "fontFamily": "Poppins-Light",
                "fontSize": 16,
                "fontWeight": "400",
                "height": 65,
                "paddingLeft": 30,
                "width": "100%",
              }
            }
            testID="inputarea-input"
          />
        </View>
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": undefined,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": "100%",
              },
              undefined,
            ]
          }
          testID="loginform-inputarea-password"
        >
          <View
            style={
              [
                {
                  "alignItems": "stretch",
                  "flex": undefined,
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "width": "100%",
                },
                {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "flex": 0,
                  "marginHorizontal": 15,
                  "paddingHorizontal": 10,
                  "top": 15,
                  "width": 95,
                  "zIndex": 1,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#363062",
                  "fontFamily": "Poppins-Light",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "textAlign": "left",
                }
              }
              testID="inputarea-label"
            >
              Password
            </Text>
          </View>
          <TextInput
            accessibilityHint="Allow to insert password"
            accessibilityLabel="Password input field"
            autoComplete="password"
            autoCorrect={false}
            onChangeText={[Function]}
            secureTextEntry={true}
            style={
              {
                "borderColor": "#363062",
                "borderRadius": 8,
                "borderWidth": 1,
                "fontFamily": "Poppins-Light",
                "fontSize": 16,
                "fontWeight": "400",
                "height": 65,
                "paddingLeft": 30,
                "width": "100%",
              }
            }
            testID="inputarea-input"
          />
        </View>
        <View
          style={
            [
              {
                "alignItems": "stretch",
                "flex": undefined,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": "100%",
              },
              {
                "marginVertical": 15,
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#363062",
                "borderColor": "#363062",
                "borderRadius": 8,
                "borderStyle": "solid",
                "height": 65,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID="loginform-button"
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontFamily": "Poppins-Bold",
                  "fontSize": 18,
                  "textAlign": "center",
                }
              }
              testID="button-text"
            >
              SIGN IN
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "alignItems": "stretch",
                "flex": undefined,
                "flexDirection": "row",
                "justifyContent": "space-around",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#595959",
                  "fontFamily": "Poppins-Light",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "textAlign": "left",
                }
              }
              testID="loginform-label"
            >
              Did you forgot your password?
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "none",
                  "borderColor": "none",
                }
              }
              testID="loginform-link"
            >
              <Text
                style={
                  {
                    "color": "#363062",
                    "fontWeight": "400",
                    "textAlign": "right",
                  }
                }
              >
                Tap here for reset
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "alignItems": "stretch",
                "flex": undefined,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": "100%",
              },
              {
                "marginBottom": 20,
                "marginTop": 50,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "marginVertical": 10,
                "width": "100%",
              }
            }
            testID="loginform-breakline"
          >
            <View
              style={
                {
                  "backgroundColor": "#BDBDBD",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                {
                  "backgroundColor": "white",
                  "color": "#595959",
                  "fontSize": 14,
                  "paddingHorizontal": 10,
                  "top": -13,
                  "zIndex": 1,
                }
              }
              testID="breakline-text"
            >
              or sign in with
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
